channels/serializers.py

```diff
@@ -160,29 +186,80 @@ def update(self, instance, validated_data):
             instance = api.update_post(post_id=post_id, text=validated_data['text'])
 
-        self._apply_vote(instance, validated_data)
+        _apply_vote(instance, validated_data)
         return api.get_post(post_id=post_id)
 
-    def _apply_vote(self, instance, validated_data):
-        """
-        Apply a vote provided by the user.
 
-        Args:
-            instance (Post): A post
-            validated_data (dict): validated data which contains the new vote from the user
+class CommentSerializer(serializers.Serializer):
+    """Serializer for comments"""
+    id = serializers.CharField(read_only=True)
+    post_id = serializers.SerializerMethodField()
+    comment_id = serializers.CharField(write_only=True, allow_blank=True, required=False)
+    text = serializers.CharField(source='body')
+    author_id = serializers.CharField(read_only=True, source='author')
+    score = serializers.IntegerField(read_only=True)
+    upvoted = WriteableSerializerMethodField()
+    created = serializers.SerializerMethodField()
+    replies = serializers.SerializerMethodField()
+
+    def get_post_id(self, instance):
+        """The post id for this comment"""
+        try:
+            # This is also available from instance.parent() but this avoids the recursive lookup
+            # which isn't necessary since we only view this from the list view, which has the
+            # post_id parameter.
+            return self.context['view'].kwargs['post_id']
+        except KeyError:
+            # This happens if we don't have the post id in the URL, for example the comment detail view
+            parent = instance.parent()
+            while not isinstance(parent, Submission):
+                parent = parent.parent()
+            return parent.id
 
-        Returns:
-            bool:
-                True if a change was made, False otherwise
-        """
-        upvote = validated_data.get('upvoted')
+    def get_upvoted(self, instance):
+        """Is a comment upvoted?"""
+        return instance.likes is True
 
-        is_upvoted = self.get_upvoted(instance)
+    def get_replies(self, instance):
+        """List of replies to this comment"""
+        return [CommentSerializer(reply, context=self.context).data for reply in instance.replies]
 
-        if upvote and not is_upvoted:
-            instance.upvote()
-        elif upvote is False and is_upvoted:
-            instance.clear_vote()
+    def get_created(self, instance):
+        """The ISO-8601 formatted datetime for the creation time"""
+        return datetime.fromtimestamp(instance.created, tz=timezone.utc).isoformat()
+
+    def validate_upvoted(self, value):
+        """Validate that upvoted is a bool"""
+        return {'upvoted': _parse_bool(value, 'upvoted')}
+
+    def create(self, validated_data):
+        api = Api(user=self.context['request'].user)
+        post_id = self.context['view'].kwargs['post_id']
+
+        kwargs = {}
+        if 'comment_id' in validated_data:
```

Comment:
edge case: noticed using the DRF web ui that it provides `''` for `comment_id`, we should probably only use this value if it has a nonempty value.